media_cdn/snippets.py

```diff
@@ -149,3 +157,38 @@ def sign_cookie(url_prefix: str, key_name: str, base64_key: str, expiration_time
             policy=policy, signature=signature)
     return signed_policy
 # [END mediacdn_sign_cookie]
+
+
+# [START mediacdn_sign_token]
+def sign_token(url_prefix: str, base64_key: bytes, algo: str, expiration_time: datetime.datetime = None) -> bytes:
+    """Gets the Signed URL Suffix string for the Media CDN' Short token URL requests.
+
+    Args:
+        url_prefix: URL prefix to sign as a string.
+        base64_key: Secret key as a base64 encoded string.
+        algo: Algorithm can be either `SHA1` or `SHA256`.
+        expiration_time: Expiration time as a UTC datetime object.
+
+    Returns:
+        Returns the Signed URL appended with the query parameters based on the
+        specified URL prefix and configuration.
+    """
+    output = b"URLPrefix=" + base64.standard_b64encode(url_prefix.encode("utf-8"))
+
+    if not expiration_time:
+        expiration_time = datetime.datetime.now() + datetime.timedelta(hours=1)
+    epoch_duration = int(
+        (expiration_time - datetime.datetime.utcfromtimestamp(0)).total_seconds()
+    )
+    output += b"~Expires=" + str(epoch_duration).encode("utf-8")
+    key = base64.standard_b64decode(base64_key)
+    algo = algo.lower()
+    if algo == "sha1":
```

Comment:
signed_tokens can also use the Ed25519 signature algorithm like other signed request modes.  In fact, we encourage people to use Ed25519 algorithm with tokens.  (We only support SHA-1 and SHA-256 for compatibility with other CDNs.)

This sample needs to support Ed25519 signatures.  The differences:
* Ed25519 uses the field `Signature` (instead of `hmac` for HMACs)
* The value is URL-safe base64 encoded (instead of hex direct for HMACs).
You can reuse the digest and signature code from the other code samples.